src/hooks/multicall.ts

```diff
@@ -7,33 +10,53 @@ export { NEVER_RELOAD } from '@uniswap/redux-multicall' // re-export for conveni
 
 // Create wrappers for hooks so consumers don't need to get latest block themselves
 
-type MulticallParams<T extends (chainId: number | undefined, latestBlock: number | undefined, ...args: any) => any> =
-  Parameters<T> extends [any, any, ...infer P] ? P : never
+interface CallState<C extends Contract, M extends keyof C['functions']> extends Omit<CallStateBase, 'result'> {
+  result: C['functions'][M] extends (...args: any) => Promise<infer T> ? T | undefined : never
+}
+
+type MultipleContractParams<
```

Comment:
Note that `*ContractParams` is actually lossy, because it loses the names/documentation around the parameters and only retains their typing. Arguably, this should be explicit so that names are retained for a better devX. wdyt?